### PR TITLE
[#230] Add `hoistMaybe` and `hoistEither` functions

### LINKED ISSUE #230: Add function to lift Maybe value into MaybeT
Problem: Currently to lift a value `ma :: Maybe a` into `MaybeT m a` you need to write `MaybeT (pure ma)`, which is bloating code with unnecessary boilerplate.

Proposed solution: Add function `hoistMaybe :: Applicative m => Maybe a -> MaybeT m a ` which already have been implemented in [`relude`](https://hackage.haskell.org/package/relude-0.7.0.0/docs/Relude-Monad-Trans.html#v:hoistMaybe). 

This will make working with MaybeT transformer more convinient

### PR DESCRIPTION
**Pull request by gpevnev**
## Description

Problem: Currently to lift a value `ma :: Maybe a` into `MaybeT m a`
you need to write `MaybeT (pure ma)`, which is bloating code with
unnecessary boilerplate.
Same idea applies for `Either e a` and `ExceptT e m a`.
`

Proposed solution:
Add functions `hoistMaybe :: Applicative m => Maybe a -> MaybeT m a` and
`hoistEither :: Applicative m => Either e a -> ExceptT e m a `

## Related issues

- Fixed #230

## ✓ Checklist for your Pull Request

Ideally a PR has all of the checkmarks set.

If something in this list is irrelevant to your PR, you should still set this
checkmark indicating that you are sure it is dealt with (be that by irrelevance).

- [x] I made sure my PR addresses a single concern, or multiple concerns which
      are inextricably linked. Otherwise I should open multiple PR's.
- [x] If your PR fixes/relates to an open issue then the description should
      reference this issue. See also [auto linking on
      github](https://help.github.com/articles/autolinked-references-and-urls/).

#### Related changes (conditional)

- Tests

  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation

  I checked whether I should update the docs and did so if necessary:

  - [x] [README](/README.md)
  - [x] Haddock

- Record your changes

  - [ ] I added an entry to the [changelog](/CHANGES.md) if my changes are visible to the users
        and
  - [ ] provided a migration guide for breaking changes if possible

#### Stylistic guide (mandatory)

- [x] My commit history is clean (only contains changes relating to my
      issue/pull request and no reverted-my-earlier-commit changes) and commit
      messages start with identifiers of related issues in square brackets.

  **Example:** `[#42] Short commit description`

  If necessary both of these can be achieved even after the commits have been
  made/pushed using [rebase and squash](https://git-scm.com/docs/git-rebase).

### CODE REVIEW

File: src/Universum/Monad/Trans.hs [hunk 1]
```diff
@@ -65,3 +69,13 @@ executingStateT s st = snd <$> usingStateT s st
...
+
+-- | Lift a 'Maybe' to the 'MaybeT' monad
+hoistMaybe  :: Applicative m => Maybe a -> MaybeT m a
+hoistMaybe m = MaybeT (pure m)
+{-# INLINE hoistMaybe #-}
+
+-- | Lift a 'Either' to the 'ExceptT' monad
+hoistEither :: Applicative m => Either e a -> ExceptT e m a
+hoistEither e = ExceptT (pure e)
+{-# INLINE hoistEither #-}
```
Comment (gpevnev) [inline src/Universum/Monad/Trans.hs:81]: @gromakovsky is this correct module to put this functionality into in your opinion?

File: src/Universum/Monad/Trans.hs [hunk 2]
```diff
@@ -14,13 +14,17 @@ module Universum.Monad.Trans
...
        ) where
 
-import Prelude (flip, fst, snd)
+import Prelude (Applicative (..), flip, fst, snd)
 
 import Universum.Functor (Functor, (<$>))
-import Universum.Monad.Reexport (Reader, ReaderT, State, StateT, runReader, runReaderT, runState,
-                                 runStateT)
+import Universum.Monad.Reexport (Either, ExceptT (..), Maybe, MaybeT (..), Reader, ReaderT, State,
+                                 StateT, runReader, runReaderT, runState, runStateT)
```
Comment (gpevnev) [inline src/Universum/Monad/Trans.hs:27]: Also, maybe I have imported something from wrong place, so be free to point on this during review

**Review by Martoon-00 — APPROVED** [review]

**Review by gromakovsky — COMMENTED** [review]
LGTM. Please add this change to the changelog and I'll approve and merge.

**Comment by gpevnev** [comment]
@gromakovsky should I add it as a new minor version (`1.7.1` for example)

**Comment by gromakovsky** [comment]
If I understand [PVP](https://pvp.haskell.org/) correctly, yes. Typically we write `Unreleased` in the changelog and replace it with the actual version when a new release is ready to be made, but in this case I suppose we can release 1.7.1 right after merging this PR.

**Review by gromakovsky — APPROVED** [review]

**Pull request merged**